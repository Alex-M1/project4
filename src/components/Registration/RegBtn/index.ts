import { connect } from 'react-redux';
import { signUpRequest, setIsRedirect } from 'store/credentials/action';
import { AppStateType } from 'store/rootReducer';
import { isRedirect } from 'store/credentials/selectors';
import RegBtn from './RegBtn';

const mapStateToProps = (state: AppStateType) => ({
  isRedirect: isRedirect(state),
});
const mapDispatchToProps = {
  signUpRequest, setIsRedirect,
};

export default connect(mapStateToProps, mapDispatchToProps)(RegBtn);
