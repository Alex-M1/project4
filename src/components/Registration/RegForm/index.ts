import { connect } from 'react-redux';
import { AppStateType } from 'store/rootReducer';
import { getInputValue } from 'store/user/selectors';
import { setCredentialsValue } from 'store/user/action';
import RegForm from './RegForm';

export interface IRegIndex {
  type: 'login' | 'password' | 'confirm';
}

const mapStateToProps = (state: AppStateType, props: IRegIndex) => ({
  value: getInputValue(state, { page: 'registration', field: props.type }),
});

const mapDispatchToProps = {
  setCredentialsValue,
};

export default connect(mapStateToProps, mapDispatchToProps)(RegForm);
