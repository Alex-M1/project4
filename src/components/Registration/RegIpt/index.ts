import { AppStateType } from 'store/rootReducer';
import { inputValue } from 'store/credentials/selectors';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { setCredentialsValue } from 'store/credentials/action';
import { IRegIndex } from './types';
import RegIpt from './RegIpt';

const mapStateToProps = (state: AppStateType, props: IRegIndex) => ({
  value: inputValue(state, 'registration', props.type),
});

const mapDispatchToProps = (dispatch: Dispatch, props: IRegIndex) => ({
  onChange: (value: string) => dispatch(setCredentialsValue({ page: 'registration', field: props.type }, value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(RegIpt);
