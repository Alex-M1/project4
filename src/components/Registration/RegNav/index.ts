import RegNav from './RegNav';

export default RegNav;
