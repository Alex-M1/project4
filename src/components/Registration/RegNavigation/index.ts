import RegNavigation from './RegNavigation';

export default RegNavigation;
