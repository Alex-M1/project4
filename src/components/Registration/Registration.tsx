import React from 'react';
import { NavLink } from 'react-router-dom';
import Header from '../Header';
import Title from '../_common_/Title';
import RegIpt from './RegIpt';
import RegBtn from './RegBtn';
import { StContainer, StGlobalCredentials, StNavLink } from '../Auth/styled';

const Registration = () => {
  return (
    <>
      <Header/>
      <StGlobalCredentials>
        <StContainer>
        <Title title="Sign Up" />
        <RegIpt
          type="login"
          placeholder="login"
        />
        <RegIpt
          type="password"
          placeholder="login"
        />
        <RegIpt
          type="confirm"
          placeholder="login"
        />
        <RegBtn />
        <StNavLink>
          <p>Already have an account?</p>
          &nbsp;
          <NavLink to="/">
            Sign In
          </NavLink>
        </StNavLink>
        </StContainer>
      </StGlobalCredentials>
    </>
  );
};

export default Registration;
