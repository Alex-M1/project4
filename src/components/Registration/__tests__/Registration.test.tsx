import React from 'react';
import configureStore from 'redux-mock-store';
import { mountSmart, shallowSmart } from 'tests/testHelper';
import Registration from '../Registration';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);

const mockStore = configureStore();
const store = mockStore({
  user: {
    auth: {
      login: '',
      password: '',
    },
    registration: {
      login: '',
      password: '',
      confirm: '',
    },
    isRedirect: false,
  },
});

describe('Registration', () => {
  it('Should match snapshot', () => {
    const component = shallowSmart(<Registration />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StGlobalCredentials', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('styled__StGlobalCredentials')).toHaveLength(1);
  });
  it('should render StContainer', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('styled__StContainer')).toHaveLength(1);
  });
  it('should render Title', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('Title')).toHaveLength(1);
  });
  it('should render RegForm', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('RegForm')).toHaveLength(3);
  });
  it('should render RegBtn', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('RegBtn')).toHaveLength(1);
  });
  it('should render RegNav', () => {
    const component = mountSmart(<Registration />, store);
    expect(component.find('RegNavigation')).toHaveLength(1);
  });
});
