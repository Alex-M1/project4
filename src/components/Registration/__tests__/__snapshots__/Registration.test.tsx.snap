// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Should match snapshot 1`] = `" <div class=\\"styled__StGlobalCredentials-dasUon cRRWpH\\"><div class=\\"styled__StContainer-rIpsz evlQxq\\"><h1 class=\\"styled__StTitle-lfcumM bhHMZN\\"></h1><div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><span class=\\"styled__StSpan-bbjaPg hFPkpt\\"></span><input type=\\"login\\" value=\\"\\" class=\\"styled__StInput-ifVcmW fkrHSd\\"/></div><div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><span class=\\"styled__StSpan-bbjaPg hFPkpt\\"></span><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW fkrHSd\\"/></div><div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><span class=\\"styled__StSpan-bbjaPg hFPkpt\\"></span><input type=\\"password\\" value=\\"\\" class=\\"styled__StInput-ifVcmW fkrHSd\\"/></div><div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><button type=\\"button\\" class=\\"styled__StButton-iwFefC jJqhBl\\"></button></div><div class=\\"styled__StNavLink-kYzesN fxjFPh\\"><p></p>Â <a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\"></a></div></div></div> "`;
